### ISSUE TITLE
Temporarily disable a redirect

### ISSUE THREAD

**Issue by harryray33** [einaregilsson/Redirector#320]
Is there a way of temporarily disabling a redirect without disabling redirector completely?

**Comment by Gitoffthelawn** [comment]
Yes.  Please open the Redirector options, click on `Edit Redirects`, and then click `Disable` for any specific redirects you desire to disable.

**Comment by harryray33** [comment]
Thanks for your reply..I know about disabling it from the options, but I was wondering if there was a way of disabling a redirect on the fly. Sometimes I just want to see the original site quickly before redirecting it without having to go to options, disabling the redirect then enabling again.

Also, when I have logging enable, where are the logs kept?

Thanks.

**Comment by harryray33** [comment]
Thanks, I've been scratching my head over this, but can't think of a way...

**Comment by endolith** [comment]
Sometimes I wish I could right-click a link and click "Open *without* redirect" without having to disable Redirector and remember to re-enable it later.

**Comment by harryray33** [comment]
Yes, same here....

**Comment by pbarquei** [comment]
Any chance holding a specific key (e.g. Shift) while clicking a link could disable the redirect rule (just for that link and that click)?
Similar to the "open link in a new tab" (hold Ctrl and click link).

**Comment by Gitoffthelawn** [comment]
> Any chance holding a specific key (e.g. Shift) while clicking a link could disable the redirect rule (just for that link and that click)? Similar to the "open link in a new tab" (hold Ctrl and click link).

That's a great idea 👍🏾 , but (some/most/all?) web browsers already have actions for shift+click, alt+click, ctrl+click defined.  Now maybe something like shift+alt+click could be possible... I'm not sure at the moment.

**Comment by endolith** [comment]
How about the right-click and context menu option?  I don't really care which method it is, but I would like at least one.

**Comment by harryray33** [comment]
Sound like good methods if they can be implemented. I find the inability to temporarily disable a redirect easily the only thing lacking in an otherwise really good and useful extension.

**Comment by Gitoffthelawn** [comment]
It's very good to learn that this is a desirable feature.

I personally find this new knowledge fascinating because it's not a feature I would likely use personally (but I'm, of course, just one person),  In the very rare situation when I need to test something because I think a redirect may be causing an issue, I quickly disable Redirector, perform the test, and then re-enable Redirector.

I'm guessing you all have more frequent situations when you want to temporarily disable a specific Redirector rule.  Would you mind giving some examples of those situations?

**Comment by pbarquei** [comment]
One situation is: take, for example, a hyperlink on a word ("PowerShell") which I would, normally, want it to take me to search for a project on github (https://github.com/search?q=PowerShell). However, if I know the project already exists, I simply want it to take me to the project (https://github.com/PowerShell).
So I would define the redirect from https://github.com/*
to
https://github.com/search?q=*
but, for example, clicking on the "hotkey" would disable the redirect.

Hope that helps.

Oh, and btw, thanks for the great job you have done with this extension.

**Comment by Hardcore-fs** [comment]
Maybe make it like "bit defender" anti virus.
you can disable with a timer, so for example with a key combo, you could pop open a window to turn of redirection for a listed site for say.. 10 minutes or 5 minutes or whatever.

**Comment by endolith** [comment]
> you can disable with a timer, so for example with a key combo

I don't like that method as well, it's not much different from disabling Redirector and then enabling it again.  Ideally this would be a 1-click quick solution for one link at a time.

**Comment by harryray33** [comment]
Another reason for me to temp disable a redirect, is that the majority of times the download site I redirect to shows me the file size and full name of the file I'm downloading.
Occasionally, when the download site doesn't show the file size info or the full file name, I like to look at the original site for the info.

If I have to do this several times it can be a bit tedious and time consuming, and, as mentioned above, I often forget to re-enable.

I like the idea of somehow temporarily disabling a particular redirect via the context menu, or maybe with a single button to toggle on or off, but presumably this would affect all redirects.

Even if a single toggle affected all redirects, it would still be preferable.
